pkg/pureutils/purefa_rest2x.go

```diff
@@ -39,3 +41,92 @@ func ListAllDestroyedVolumesFromFA(faClient *flasharray.Client) ([]flasharray.Vo
 	}
 	return volumes, nil
 }
+
+func ListAllRealmsFromFA(faClient *flasharray.Client) ([]flasharray.RealmResponse, error) {
+	params := make(map[string]string)
+	params["destroyed"] = "false"
+	realms, err := faClient.Realms.ListAllAvailableRealms(params)
+	if err != nil {
+		return nil, err
+	}
+	return realms, nil
+}
+
+func ListAllPodsFromFA(faClient *flasharray.Client) ([]flasharray.PodResponse, error) {
+	params := make(map[string]string)
+	params["destroyed"] = "false"
+	pods, err := faClient.Pods.ListAllAvailablePods(params)
+	if err != nil {
+		return nil, err
+	}
+	return pods, nil
+}
+
+func CreatePodinFA(faClient *flasharray.Client, podName string) (*[]flasharray.PodResponse, error) {
+	queryParams := make(map[string]string)
+	queryParams["names"] = fmt.Sprintf("%s", podName)
+	podinfo, err := faClient.Pods.CreatePod(queryParams, nil)
+	if err != nil {
+		return nil, err
+	}
+	return podinfo, nil
+}
+
+func DeletePodinFA(faClient *flasharray.Client, podName string) error {
+	queryParams := make(map[string]string)
+	queryParams["names"] = fmt.Sprintf("%s", podName)
+	queryParams["destroy_contents"] = "true"
+	data := map[string]bool{"destroyed": true}
+
+	deleteParams := make(map[string]string)
+	deleteParams["names"] = fmt.Sprintf("%s", podName)
+	deleteParams["eradicate_contents"] = "true"
+
+	err := faClient.Pods.DeletePod(queryParams, deleteParams, data)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+func GetFARealmFromMgmtEndpoint(secret PXPureSecret, mgmtEndPoint string) string {
+	for _, faDetails := range secret.Arrays {
+		if faDetails.MgmtEndPoint == mgmtEndPoint {
+			return faDetails.Realm
+		}
+	}
+	return ""
+}
+func IsFARealmExistsOnMgmtEndpoint(faClient *flasharray.Client, realm string) (bool, error) {
+	params := make(map[string]string)
+	params["destroyed"] = "false"
+	realms, err := faClient.Realms.ListAllAvailableRealms(params)
+	if err != nil {
+		return false, err
+	}
+	for _, eachRealmItems := range realms {
+		for _, eachRealm := range eachRealmItems.Items {
+			if strings.Contains(eachRealm.Name, realm) {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+func IsPodExistsOnMgmtEndpoint(faClient *flasharray.Client, podName string) (bool, error) {
```

Comment:
update comment on what the function does

Reply:
done